### PR TITLE
Add hello examples with new APIs

### PR DESCRIPTION
**Pull request by nvkevlu**
Add hello examples with new APIs

### Description

Add hello examples with new job and client APIs, re-based version of this: https://github.com/NVIDIA/NVFlare/pull/2723 but without deleting the previous hello examples for now. We need the current ones for CI, and we can still reorganize this.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

**Comment by YuanTingHsieh** [comment]
Because these are new styles of example that currently are not supported in CI.

I could just move all old hello examples to testing directory and change CI to that dir for testing.
So it does not confuses users.
Users will only see new style of examples.

**Comment by nvkevlu** [comment]
> Because these are new styles of example that currently are not supported in CI.
> 
> I could just move all old hello examples to testing directory and change CI to that dir for testing. So it does not confuses users. Users will only see new style of examples.

I think that sounds good. I was looking into the CI and are we actually testing from hello examples from the examples directory or just using the configurations under tests/integration_test/data?

**Comment by nvkevlu** [comment]
I decided to move and keep the existing CI jobs out of the way in test/integration_test/data/test_configs/standalone_job/previous_jobs, and for the new Job API jobs, I did not export jobs because I thought it would create too much extra clutter in the beginning that may confuse and intimidate new users, and if they are ready to look into the job, they can run the export job command themself when they are ready to look at it.

File: examples/hello-world/hello-pt/README.md
Comment (YuanTingHsieh) [inline examples/hello-world/hello-pt/README.md:46]: We maybe need another example to explain this, don't need to be in hello example though

File: examples/hello-world/hello-fedavg-numpy/src/hello-numpy_fl.py
Comment (YuanTingHsieh) [inline examples/hello-world/hello-fedavg-numpy/src/hello-numpy_fl.py:48]: I think Initial "Params" should not come from the client side.

File: examples/hello-world/hello-fedavg-numpy/requirements.txt
Comment (YuanTingHsieh) [inline examples/hello-world/hello-fedavg-numpy/requirements.txt:1]: Should be 2.5.0rc

File: examples/hello-world/hello-cyclic/README.md
Comment (YuanTingHsieh) [inline examples/hello-world/hello-cyclic/README.md:55]: Do users have to export?
What if they just specify this flag when running the "python3 cyclic_script-executor-hello-cyclic.py"?

File: examples/hello-world/hello-cross-val/jobs/hello-cross-val/app/custom/train.py
Comment (YuanTingHsieh) [inline examples/hello-world/hello-cross-val/jobs/hello-cross-val/app/custom/train.py:1]: If this is new file may need to be 2024?

File: examples/hello-world/hello-cross-val/jobs/hello-cross-val/app/custom/net.py
Comment (YuanTingHsieh) [inline examples/hello-world/hello-cross-val/jobs/hello-cross-val/app/custom/net.py:1]: Should this be 2024?

File: docs/examples/hello_pt_job_api.rst
Comment (YuanTingHsieh) [inline docs/examples/hello_pt_job_api.rst:54]: Do we know What is this issue?

File: docs/examples/hello_tf_job_api.rst
Comment (YuanTingHsieh) [inline docs/examples/hello_tf_job_api.rst:111]: Do we want to mention DXO and Shareable here where are we showing them ?

**Comment by YuanTingHsieh** [comment]
Great PR with a lot of details, added some comments

File: docs/examples/hello_pt_job_api.rst
Comment (nvkevlu) [reply docs/examples/hello_pt_job_api.rst:54]: This was copied over from the previous example, I tried to look into it briefly but did not find good specifics on this, perhaps don't need this anymore?

File: docs/examples/hello_tf_job_api.rst
Comment (nvkevlu) [reply docs/examples/hello_tf_job_api.rst:111]: I copied this from the previous example without updating, I will update this

File: tests/integration_test/data/test_configs/standalone_job/hello_numpy_examples.yml
Comment (nvkevlu) [reply tests/integration_test/data/test_configs/standalone_job/hello_numpy_examples.yml:8]: I do not have that one in the new apis yet

File: examples/hello-world/hello-tf/src/hello-tf_fl.py
Comment (nvkevlu) [reply examples/hello-world/hello-tf/src/hello-tf_fl.py:46]: I removed the prepare_data.sh but I'm not sure what to clarify or what do you mean by the dataset should be disjoint already

File: docs/examples/hello_cross_site_eval.rst
Comment (nvkevlu) [reply docs/examples/hello_cross_site_eval.rst]: updated the wording to make this more general

File: examples/hello-world/hello-pt/README.md
Comment (nvkevlu) [reply examples/hello-world/hello-pt/README.md:46]: Yes, this relates to the comment you made on hello-numpy initial params from client side, I did try but had issues with Job API to do it with the existing components, so this should be showcased in another example. After this PR in a new one perhaps.

File: examples/hello-world/hello-cross-val/jobs/hello-cross-val/app/custom/train.py
Comment (nvkevlu) [reply examples/hello-world/hello-cross-val/jobs/hello-cross-val/app/custom/train.py]: updated

File: examples/hello-world/hello-cross-val/jobs/hello-cross-val/app/custom/net.py
Comment (nvkevlu) [reply examples/hello-world/hello-cross-val/jobs/hello-cross-val/app/custom/net.py]: updated

File: examples/hello-world/hello-tf/src/hello-tf_fl.py
Comment (YuanTingHsieh) [reply examples/hello-world/hello-tf/src/hello-tf_fl.py:46]: I mean in reality different site has different dataset, otherwise defeat the purpose of FL.

So this "if site-1 read first half and if site-2 read second half" scenario does not fit into reality.

This is OK in hello-examples, just wondering if we need to clarify this.

File: examples/hello-world/hello-cyclic/README.md
Comment (YuanTingHsieh) [reply examples/hello-world/hello-cyclic/README.md:55]: Instead of asking users to use job api to export then run this nvflare simulator command,

Can it be just one liner? `TF_FORCE_GPU_ALLOW_GROWTH=true  python3 cyclic_script-executor-hello-cyclic.py`

File: examples/hello-world/hello-tf/src/hello-tf_fl.py
Comment (nvkevlu) [reply examples/hello-world/hello-tf/src/hello-tf_fl.py:46]: I see, there is just a single line just mentioning that it is split in half in the doc right now, I could add a little more

File: examples/hello-world/hello-cyclic/README.md
Comment (nvkevlu) [reply examples/hello-world/hello-cyclic/README.md]: Thanks, updated it to simplify it. I'm not sure about the section for so removed it for now:

If you possess more GPUs than clients,
an alternative strategy is to run one client on each GPU.

Comment (YuanTingHsieh) [reply examples/hello-world/hello-cyclic/README.md]: yeah that statement is not 100% solving this issue, we can remove that

File: docs/examples/hello_pt_job_api.rst
Comment (YuanTingHsieh) [reply docs/examples/hello_pt_job_api.rst:54]: @SYangster @IsaacYangSLA @holgerroth do you remember anything regarding this?

File: examples/hello-world/hello-fedavg-numpy/src/hello-numpy_fl.py
Comment (nvkevlu) [reply examples/hello-world/hello-fedavg-numpy/src/hello-numpy_fl.py:48]: I tried to make it from the server side but could not get it to work with the existing components. It may require some more fixing than the scope of this PR, could we change it in a future PR?

File: examples/hello-world/hello-fedavg-numpy/requirements.txt
Comment (nvkevlu) [reply examples/hello-world/hello-fedavg-numpy/requirements.txt:1]: This needs to be updated everywhere so maybe we do it in a PR just for updating the version of this?

File: docs/examples/hello_pt_job_api.rst
Comment (YuanTingHsieh) [reply docs/examples/hello_pt_job_api.rst:54]: ok, I see this traces back to long time ago, maybe we just remove these sections.

Comment (nvkevlu) [reply docs/examples/hello_pt_job_api.rst]: removed

**Comment by nvkevlu** [comment]
CI doesn't pass because we are adding new pages, and the link checker does not consider pages being added as existing

**Review by YuanTingHsieh — APPROVED** [review]

**Comment by YuanTingHsieh** [comment]
Ok we can merge this then make sure ci passing after this merge

**Pull request merged**
